prboom2/src/gl_missingtexture.c

```diff
@@ -111,6 +159,56 @@ static void gld_PrepareSectorSpecialEffects(void)
   }
 }
 
+static void gld_RegisterBleedthroughSector(sector_t* source, sector_t* target, int ceiling)
+{
+    int i;
+    int source_idx = -1;
+    assert(source);
+    assert(target);
+
+    /* check whether the sector is processed already */
+    for (i = 0; i < numbleedsectors && source_idx == -1; i++)
+        if (bleedsectors[i].source == source && bleedsectors[i].ceiling == ceiling)
+            source_idx = i;
+    
+    if (source_idx == -1) {
+        /* allocate memory for new sector */
+        if (!bleedsectors) {
```

Comment:
There's an easier more generic solution that considers the fact that `realloc(NULL, ...)` is equal to `malloc(...)`. But that's mostly a matter of style.

Reply:
Ah, yes. I thought this might be nicer for readability but I can do it that way too.

Reply:
I see it the other way round, I find it easier to read if this just uses plain `realloc()`.

Reply:
Sure, addressed this in the latest push.